src/abstractdataframe/abstractdataframe.jl

```diff
@@ -417,22 +416,32 @@ missing values, but does not currently contain any.
 ```julia
 df = DataFrame(i = 1:10, x = rand(10), y = rand(["a", "b", "c"], 10))
 describe(df)
+describe(df, stats = :all)
+describe(df, stats = [:min, :max])
 ```
 
 """
-StatsBase.describe(df::AbstractDataFrame; kwargs...) = describe(stdout, df; kwargs...)
-function StatsBase.describe(io::IO, df::AbstractDataFrame; stats::AbstractVector{Symbol} = 
-                            [:mean, :min, :median, :max, :nmissing, :eltype])
+function StatsBase.describe(df::AbstractDataFrame; stats::Union{Symbol,AbstractVector{Symbol}} = 
+                            [:mean, :min, :median, :max, :nunique, :nmissing, :eltype])
     # Check that people don't specify the wrong fields. 
     allowed_fields = [:mean, :std, :min, :q25, :median, :q75, 
-                      :max, :nunique, :nmissing, :eltype] 
-    if !issubset(stats, allowed_fields) 
+                      :max, :nunique, :nmissing, :first, :last, :eltype] 
+    
+    if typeof(stats) == Symbol && stats != :all 
```

Comment:
This throws error when a call like `describe(df, stats=:mean)` is passed. I thought we want to allow passing a single stat not inside a vector. Also The printed error looks e.g. like:

```
ERROR: ArgumentError: Field(s) not allowed: mean. Allowed fields are: Symbol[:mean, :std, :min, :q25, :median, :q75, :max, :nunique, :nmissing, :first, :last, :eltype], or `stats = :all`
```
I would not print `Symbol` in front of the bracket (but it is a minor issue).

Additionally in this case you know that one field was passed so `Field not allowed:` would be better correct (also a minor issue).

If we do not want to allow passing a single symbol then I would recommend to make an error message clearer that the passed list of fields must be a vector. Now the error message is not very clear.

Reply:
No matter what functionality is finally implemented I would test passing a single `Symbol`, e.g. `:mean`, as `stats` (to test that it either throws an error or produces a correct result).

Reply:
I was under the impression it was either a a vector `[:mean, :min, :max...]` or the symbol `:all`. 

If we want `stats = :mean` to be allowed, we could add something like 

`isa(stats, Symbol) ? stats = [stats] : nothing`

Reply:
For me it is OK to require a vector or `:all` but then I recommend to make the error message clearer.

The fix should rather be to add after line 430 something like:

```
if stats in allowed_fields
    stats = [stats]
else
... here go on with the error message we currently have
end
```

Reply:
I think I will do the above, because if someone just wants the means of all their columns, it makes sense for them to write `stats = :mean`. An alternative way to do that is a bit more cumbersome.

Reply:
Here is what I have now. I try to have two separate error messages, for if someone put in the wrong symbol and if they put in the wrong vector of symbols. 

With regards to the awkward `Symbol[:whatever]` in the error message, this is because of the escaping `$`. I'm not sure how to avoid this without writing it out manually, which I can do obviously. 

Should I push this change and then you can tweak the error message if you want? 

```
    # Check that people don't specify the wrong fields. 
    allowed_fields = [:mean, :std, :min, :q25, :median, :q75, 
                      :max, :nunique, :nmissing, :first, :last, :eltype] 
    if stats == :all
        stats = allowed_fields 
    end

    if typeof(stats) == Symbol 
        if !(stats in allowed_fields)
            throw(ArgumentError("$stats not allowed. Allowed fields are " * 
                ":mean, :std, :min, :q25, :median, :q75, " *
                ":max, :nunique, :nmissing, :first, :last, :eltype"))
        else 
            stats = [stats]
        end
    end

    if !issubset(stats, allowed_fields)
        disallowed_fields = setdiff(stats, allowed_fields)
        not_allowed = "Field(s) not allowed: $disallowed_fields. "
        allowed = "Allowed fields are: $allowed_fields"
        throw(ArgumentError(not_allowed * allowed)) 
    end
```

Reply:
Great. As for error message I would go for something like:

```
 # Check that people don't specify the wrong fields. 
    allowed_fields = [:mean, :std, :min, :q25, :median, :q75, 
                      :max, :nunique, :nmissing, :first, :last, :eltype]
    allowed_msg = "\nAllowed fields are: :" * join(allowed_fields, ", :")
    if stats == :all
        stats = allowed_fields 
    end

    if typeof(stats) == Symbol 
        if !(stats in allowed_fields)
            throw(ArgumentError("$stats not allowed." * allowed_msg))
        else 
            stats = [stats]
        end
    end

    if !issubset(stats, allowed_fields)
        disallowed_fields = setdiff(stats, allowed_fields)
        not_allowed = "Field(s) not allowed: :" * join(disallowed_fields, ", :") * "."
        throw(ArgumentError(not_allowed * allowed_msg)) 
    end
```
does it look good to you?

Reply:
Addded and pushed. I add a ": " somewhere so the error message reflects a symbol.